models/CommentManager.php

```diff
@@ -0,0 +1,132 @@
+            $req = $this->db->query("
+                SELECT  comments.id,
+                        comments.name,
+                        comments.email,
+                        comments.date,
+                        comments.article_id,
+                        comments.comment,
+                        articles.title
+                FROM comments
+                JOIN articles
+                ON comments.article_id = articles.id
+                WHERE comments.seen = '0'
+                ORDER BY comments.date ASC
+            ");
+
+            $results = [];
+            while($rows = $req->fetchObject()){
+                $results[] = $rows;
+            }
+            return $results;
+        }
+
+        // Fonction permettant de supprimer un commentaire
+        function delete_comment(){
+            $delete = filter_input(INPUT_GET, 'delete', FILTER_SANITIZE_NUMBER_INT);
+
+            $req = $this->db->prepare('DELETE FROM comments WHERE id = ?');
+            $req->execute(array($delete));
+
+            header("Location: /dashboard");
+        }
+
+        // Fonction permettant d'approuver un commentaire
+        function approve_comment(){
+            $id = filter_input(INPUT_GET, 'approve', FILTER_SANITIZE_NUMBER_INT);
+
+            $query = $this->db->prepare('UPDATE comments SET seen=1 WHERE id=:id');
+
+            $query->execute(['id' => $id]);
+
+            header("Location: /dashboard");
+        }
+
+
+        // Fonction permettant de vérifier que les données envoyées pat l'utilisateur sont bonnes.
+        function form_comment_verification(){
+            if(filter_has_var(INPUT_POST, 'submit')){
+                $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name')));
+
+                $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email')));
+
+                $comment = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'comment')));
+                $errors = [];
+
+                if(empty($name) || empty($email) || empty($comment)){
+                    $errors['empty'] = "Tous les champs n'ont pas été remplis";
+                }else{
+                    if(!filter_var($email, FILTER_VALIDATE_EMAIL)){
+                        $errors['email'] = "L'adresse email n'est pas valide";
+                    }
+                    if(strlen($name) < 5){
+                        $errors['name'] = "Votre nom doit contenir au moins 5 caractères.";
+                    }
+                    if(strlen($comment) < 5){
+                        $errors['comment'] = "Votre message doit contenir au moins 5 caractères.";
+                    }
+                }
+
+                if(!empty($errors)){
+                ?>
+                    <div class="card red">
+                        <div class="card-content white-text">
+                            <?php
+                                foreach($errors as $error){
+                                    echo $error."</br>";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of echo language construct is discouraged.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)